integrations/openai/src/databricks_openai/agents/session.py

```diff
@@ -0,0 +1,329 @@
+            **pool_kwargs: Additional arguments passed to LakebasePool.
+        """
+        self.session_id = session_id
+        self.sessions_table = sessions_table
+        self.messages_table = messages_table
+
+        self._pool = _get_or_create_pool(
+            instance_name=instance_name,
+            workspace_client=workspace_client,
+            **pool_kwargs,
+        )
+
+        if not self._tables_exist():
+            self._create_tables()
+
+        self._ensure_session()
+
+    def _tables_exist(self) -> bool:
+        """Check if both session tables already exist."""
+        with self._pool.connection() as conn:
+            result = conn.execute(
+                """
+                SELECT COUNT(*) as cnt FROM information_schema.tables 
+                WHERE table_name IN (%s, %s)
+                """,
+                (self.sessions_table, self.messages_table),
+            )
+            row = result.fetchone()
+            # Both tables should exist
+            return row["cnt"] == 2
+
+    def _create_tables(self) -> None:
+        """Create the required tables. Should only be called by the first user/admin."""
+        sessions_sql = self.CREATE_SESSIONS_TABLE_SQL.format(sessions_table=self.sessions_table)
+        messages_sql = self.CREATE_MESSAGES_TABLE_SQL.format(
+            sessions_table=self.sessions_table, messages_table=self.messages_table
+        )
+
+        with self._pool.connection() as conn:
+            conn.execute(sessions_sql)
+            conn.execute(messages_sql)
+
+        logger.info(f"Created tables {self.sessions_table}, {self.messages_table}")
+
+    def _ensure_session(self) -> None:
+        """Ensure the session record exists in agent_sessions table."""
+        now = datetime.now(timezone.utc)
+
+        with self._pool.connection() as conn:
+            conn.execute(
+                f"""
+                INSERT INTO {self.sessions_table} (session_id, created_at, updated_at)
+                VALUES (%s, %s, %s)
+                ON CONFLICT (session_id) DO NOTHING
+                """,
+                (self.session_id, now, now),
+            )
+
+        logger.debug(f"Ensured session {self.session_id} exists")
+
+    async def get_items(self, limit: int | None = None) -> list[TResponseInputItem]:
+        """
+        Retrieve the conversation history for this session.
+
+        Args:
+            limit: Maximum number of items to retrieve. If None, retrieves all items.
+                   When specified, returns the latest N items in chronological order.
+
+        Returns:
+            List of input items representing the conversation history.
+        """
+        if limit is not None:
+            # Get the last N items, but return in chronological order
+            query = f"""
+                SELECT message_data FROM (
+                    SELECT message_data, id
+                    FROM {self.messages_table}
```

Comment:
Let's not do plain string formatting here. It could have security implications.

Use [`psycopg.sql.Identifier`](https://www.psycopg.org/psycopg3/docs/api/sql.html#psycopg.sql.Identifier) to validate the table name